### PR TITLE
Add Tron block API

### PR DESCRIPTION
**Pull request by Pantani**
- Add block API support for TRON
- Refactor TRON API
- Split stake stuffs in a separate file

closes https://github.com/trustwallet/blockatlas/issues/209

### CODE REVIEW

**Comment by vikmeup** [comment]
Does this support only TRX transfers but also TRC10?

**Comment by Pantani** [comment]
> Does this support only TRX transfers but also TRC10?


Only TRX, I going see how to filter the TRC10 txs

File: platform/tron/api.go [hunk 1]
```diff
@@ -27,6 +27,32 @@ func (p *Platform) Coin() coin.Coin {
 	return coin.Coins[coin.TRX]
 }
 
+func (p *Platform) CurrentBlockNumber() (int64, error) {
+	return p.client.CurrentBlockNumber()
+}
+
+func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
```
Comment (dpereskokov) [inline platform/tron/api.go]: do we need pointer here?

File: platform/tron/api.go [hunk 2]
```diff
@@ -82,10 +82,9 @@ func (p *Platform) GetTokenTxsByAddress(address, token string) (blockatlas.TxPag
 
 	var tokenInfo AssetInfo
 	info, err := p.client.GetTokenInfo(token)
-	if err != nil && len(info.Data) == 0 {
+	if err != nil || len(info.Data) == 0 {
```
Comment (vikmeup) [inline platform/tron/api.go]: is this intended? it might cause issues

File: platform/tron/stake.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
+package tron
+
+import (
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+	"github.com/trustwallet/blockatlas/pkg/logger"
+	services "github.com/trustwallet/blockatlas/services/assets"
```
Comment (vikmeup) [inline platform/tron/stake.go]: I like this idea to keep in separate file as `tron/stake.go` for each interface, makes it clean

Comment (Pantani) [reply platform/tron/stake.go]: yes, I did start the refactor of the API, but I'm starting to be crazy with a lot of methods the `api.go` has. I did split the stake things to be more clear, we can do that for every API

File: platform/tron/api.go (hunk 1 above)
Comment (Pantani) [reply platform/tron/api.go]: yes! to satisfy the platform methods inside the registry. And this pointer going to stay at a high level, don't have a chance to the garbage collector can't desalloc that

File: platform/tron/api.go (hunk 2 above)
Comment (Pantani) [reply platform/tron/api.go]: yes, the old logic doesn't make a lot of sense

**Comment by Pantani** [comment]
> Does this support only TRX transfers but also TRC10?

@vikmeup now support!!!!

**Pull request merged**